logic/logic.go

```diff
@@ -15,14 +14,14 @@ type Resulter int
 
 // ResulterFunc is a function that will compute skills for the referenced
 // Result. It will call Taskers of encapsulated tasks on demand.
-type ResulterFunc func(ctx context.Context, resultKey *datastore.Key) error
+type ResulterFunc func(ctx context.Context, result model.KeyedResult, challenge model.Challenge) error
 
 // Tasker is a reference to a unique implementation of a TaskerFunc.
 type Tasker int
 
 // TaskerFunc is a function that will compute task results for the given Task
 // and User.
-type TaskerFunc func(ctx context.Context, task, result, user *datastore.Key) (model.Skills, error)
+type TaskerFunc func(ctx context.Context, task model.KeyedTask, result model.KeyedResult, user model.User, challenge model.Challenge) (model.Skills, error)
```

Comment:
we need the challenge to get the task index to know what time to get from the result.